pymc/aesaraf.py

```diff
@@ -893,11 +897,58 @@ def local_check_parameter_to_ninf_switch(fgraph, node):
 )
 
 
+def find_rng_nodes(variables: Iterable[Optional[TensorVariable]]):
+    """Return RNG variables in a graph"""
+    return [
+        node
+        for node in graph_inputs(variables)
+        if isinstance(
+            node,
+            (
+                at.random.var.RandomStateSharedVariable,
+                at.random.var.RandomGeneratorSharedVariable,
+            ),
+        )
+    ]
+
+
+def reseed_rngs(rngs: Iterable[SharedVariable], seed: "RandomSeed") -> None:
```

Comment:
This comment came after looking at `compile_pymc`. When no random seed is provided, and `seed is None`. What do you want this to do exactly?

Reply:
When no seed is provided we still want to override the values. This is because by default when you initialize a Aesara RandomVariable without specifying a RandomState/Shared variable it will create one on the spot. This however will not have the nice properties of Seeding with a SeedSequence and there's a considerable high change of collision/ bad bit_generators.

Reply:
I also thought about using a sentiel default value `UNSET`, and not do anything when a user explicitly passes `None`.

Reply:
Another option that I just realized is that we could play around with the extra keyword arguments of `SeedSequence`:

```python
>>> np.random.SeedSequence(entropy=1).spawn(1)
SeedSequence(
    entropy=1,
    spawn_key=(1,),
)
>>> np.random.SeedSequence(entropy=1, n_children_spawned=1).spawn(1)[0]
SeedSequence(
    entropy=1,
    spawn_key=(1,),
)
>>> np.random.SeedSequence(entropy=1, spawn_key=[1])
SeedSequence(
    entropy=1,
    spawn_key=(1,),
)
>>> np.random.SeedSequence(entropy=1, spawn_key=[1]).spawn(1)[0]
SeedSequence(
    entropy=1,
    spawn_key=(1, 0),
)
```

It looks like there's a bunch of ways to get the exact same `SeedSequence`, and that if you use a `SeedSequence` that already has a `spawn_key`, the result is a new `SeedSequence` that can generate independent streams from its parent or siblings.